src/handlers/issue/generate-nft-mint-signature.ts

```diff
@@ -0,0 +1,127 @@
+import { BigNumber, BigNumberish, ethers, utils } from "ethers";
+import { getPayoutConfigByNetworkId } from "../../helpers/payout";
+import { MaxUint256 } from "@uniswap/permit2-sdk";
+import { keccak256, toUtf8Bytes } from "ethers/lib/utils";
+
+const NFT_MINTER_PRIVATE_KEY = process.env.NFT_MINTER_PRIVATE_KEY as string;
+const NFT_CONTRACT_ADDRESS = process.env.NFT_CONTRACT_ADDRESS as string;
+const NFT_CONTRACT_CHAIN_ID = parseInt(process.env.NFT_CONTRACT_CHAIN_ID as string);
+const SIGNING_DOMAIN_NAME = "NftReward-Domain";
+const SIGNING_DOMAIN_VERSION = "1";
+
+interface NftMintRequest {
+  beneficiary: string;
+  deadline: BigNumberish;
+  keys: string[];
+  nonce: BigNumberish;
+  values: string[];
+}
+
+const domain = {
+  name: SIGNING_DOMAIN_NAME,
+  version: SIGNING_DOMAIN_VERSION,
+  verifyingContract: NFT_CONTRACT_ADDRESS,
+  chainId: NFT_CONTRACT_CHAIN_ID,
+};
+
+const types = {
+  MintRequest: [
+    { name: "beneficiary", type: "address" },
+    { name: "deadline", type: "uint256" },
+    { name: "keys", type: "bytes32[]" },
+    { name: "nonce", type: "uint256" },
+    { name: "values", type: "string[]" },
+  ],
+};
+
+const keys = [
+  "GITHUB_ORGANIZATION_NAME",
+  "GITHUB_REPOSITORY_NAME",
+  "GITHUB_ISSUE_ID",
+  "GITHUB_USERNAME",
+  "GITHUB_CONTRIBUTION_TYPE",
+];
+
+interface NftMintRequestData {
+  request: {
+    beneficiary: string;
+    deadline: string;
+    keys: string[];
+    nonce: string;
+    values: string[];
+  };
+  nftMetadata: Record<string, string>;
+  nftAddress: string;
+  networkId: number;
+  signature: string;
+}
+
+type GenerateNftMintSignatureParams = {
+  organizationName: string;
+  repositoryName: string;
+  issueNumber: string;
+  beneficiary: string;
+  username: string;
+  contributionType: string;
+};
+
+export async function generateNftMintSignature({
+  organizationName,
+  repositoryName,
+  issueNumber,
+  beneficiary,
+  username,
+  contributionType,
+}: GenerateNftMintSignatureParams) {
+  const { rpc } = getPayoutConfigByNetworkId(NFT_CONTRACT_CHAIN_ID);
+
+  let provider;
+  let adminWallet;
+  try {
+    provider = new ethers.providers.JsonRpcProvider(rpc);
+  } catch (error) {
+    throw console.error("Failed to instantiate provider", error);
+  }
+
+  try {
+    adminWallet = new ethers.Wallet(NFT_MINTER_PRIVATE_KEY, provider);
+  } catch (error) {
+    throw console.error("Failed to instantiate wallet", error);
+  }
+
+  const nftMintRequest: NftMintRequest = {
+    beneficiary: beneficiary,
+    deadline: MaxUint256,
+    keys: keys.map((key) => utils.keccak256(utils.toUtf8Bytes(key))),
+    nonce: BigNumber.from(keccak256(toUtf8Bytes(`${organizationName}${repositoryName}${issueNumber}${username}`))),
```

Comment:
I vaguely recall that we have the values delimitated with hyphens. This won't affect runtime behavior in any way but code consistency is nice. Let's be sure to match styles if there is any differentiation. 

```suggestion
    nonce: BigNumber.from(keccak256(toUtf8Bytes(`${organizationName}-${repositoryName}-${issueNumber}-${username}`))),
```

Reply:
Hey @whilefoo it looks like we are only using the user id which doesn't seem right to me. I thought that we at least encode the issue ID? 

https://github.com/ubiquibot/comment-incentives/blob/7b88ed93321d902cf6e7f41cc839ccf105dedbdf/src/handlers/issue/generate-permit-2-signature.ts#L50

@wannacfuture rfc